source/addonStore/dataManager.py

```diff
@@ -162,6 +169,49 @@ def _cacheCompatibleAddons(self, addonData: str, cacheHash: Optional[str]):
 		with open(self._cacheCompatibleFile, "w", encoding="utf-8") as cacheFile:
 			json.dump(cacheData, cacheFile, ensure_ascii=False)
 
+	def _shouldCacheCompatibleAddonsBackup(self) -> bool:
+		if not os.path.exists(self._cacheCompatibleOldFile):
+			return True
+		resetNewAddons = config.conf["addonStore"]["resetNewAddons"]
+		if resetNewAddons == "startup":
+			return True
+		lastBackupTime = os.path.getmtime(self._cacheCompatibleOldFile)
+		lastBackupDate = datetime.fromtimestamp(lastBackupTime)
+		nowDate = datetime.now()
+		diffDate = nowDate - lastBackupDate
+		if resetNewAddons == "weekly" and diffDate.days >= 7:
+			return True
+		if resetNewAddons == "monthly" and diffDate.days >= 30:
+			return True
+		return False
+
+	def _getResetNewAddonsDate(self) -> str:
+		resetNewAddons = config.conf["addonStore"]["resetNewAddons"]
+		if resetNewAddons == "startup":
+			# Translators: Message presented in the new add-ons tab, informing that new add-ons will be reset at startup
+			return _("Will be reset at startup")
+		lastBackupTime = os.path.getmtime(self._cacheCompatibleOldFile)
```

Comment:
I haven't read the code, but you should be catching a file not found error here I believe.
On a new portable copy of this PR's artifact, I'm getting:

```
Traceback (most recent call last):
  File "gui\blockAction.pyc", line 79, in funcWrapper
  File "gui\__init__.pyc", line 434, in onAddonStoreCommand
  File "gui\blockAction.pyc", line 79, in funcWrapper
  File "gui\__init__.pyc", line 237, in popupSettingsDialog
  File "gui\addonStoreGui\controls\storeDialog.pyc", line 56, in __init__
  File "gui\settingsDialogs.pyc", line 225, in __init__
  File "gui\addonStoreGui\controls\storeDialog.pyc", line 94, in makeSettings
  File "gui\addonStoreGui\controls\storeDialog.pyc", line 205, in _createFilterControls
  File "gui\addonStoreGui\controls\storeDialog.pyc", line 205, in <genexpr>
  File "utils\displayString.pyc", line 62, in displayString
  File "addonStore\models\status.pyc", line 53, in _displayStringLabels
  File "addonStore\models\status.pyc", line 457, in getResetNewAddonsDate
  File "addonStore\dataManager.pyc", line 193, in _getResetNewAddonsDate
  File "<frozen genericpath>", line 55, in getmtime
FileNotFoundError: [WinError 2] The system cannot find the file specified: 'C:\\Users\\luke\\Desktop\\NVDA versions\\pr4\\userConfig\\addonStore\\_cachedCompatibleAddons-old.json'
```